pkg/apis/configuration/v1/kongprotocol_types.go

```diff
@@ -0,0 +1,31 @@
+package v1
+
+//+ KongProtocol is a valid Kong protocol
+//+ This alias is necessary to deal with https://github.com/kubernetes-sigs/controller-tools/issues/342
+//+kubebuilder:validation:Enum=http;https;grpc;grpcs;tcp;tls;udp
+//+kubebuilder:object:generate=true
+type KongProtocol string
+
+func KongProtocolsToStrings(protocols []KongProtocol) (res []string) {
+	for _, protocol := range protocols {
+		res = append(res, string(protocol))
+	}
+	return
+}
+
+func StringsToKongProtocols(strings []string) (res []KongProtocol) {
```

Comment:
lint: public symbols should have godoc comments

Reply:
07d8f6a3